index.js

```diff
@@ -27,7 +27,8 @@ const defaultOpts = {
 module.exports = function(opts = {}) {
   opts = { ...defaultOpts, ...opts };
 
-  opts.loader = new Preprocessor(opts.loader, ".").process(opts);
+  const { loader, ...defines } = opts;
+  opts.loader = new Preprocessor(opts.loader, ".").process(defines);
```

Comment:
Fun fact: if you pass in an option whose value contains `#put`, all hell breaks loose. Took me a while to figure it out. Removing `loader` from the options that are passed into `.process(...)` fixes it.

Reply:
I need to find a better plugin anyways. This one triggers the security audits.